src/lib.rs

```diff
@@ -390,10 +407,16 @@ fn command<'a>(i: &'a [u8]) -> IResult<&'a [u8], Command<'a>> {
     )
 }
 
+/// An error as returned by a `Handler` method.
 pub enum Error {
-    // The meaning of the value is not defined by the protocol; so it
-    // can be used by a handler for debugging.
+    /// A plain error.  The meaning of the value is not defined by the
+    /// protocol; so it can be used by a handler for debugging.
     Error(u8),
+    /// The request is not implemented.  Note that, in some cases, the
+    /// protocol implementation claims that a feature is implemented;
+    /// if the protocol then returns `Unimplemented`, the client will
+    /// be confused.  So, normally it is best either to not implement
+    /// a `Handler` method, or to return `Error` from implementations.
```

Comment:
This comment was a bit confusing, I think due to the overloading of "protocol" in the second sentence.  Are you trying to say that `Unimplemented` is essentially reserved to the library and shouldn't be used in user-specific `Handler` implementations?

Reply:
I meant that `Handler` implementation should not return this.

However there is a design question here.  For example we could either have handlers specify the features they implement (and then reply definitively to `qSupported`), or we could change some `qSupported` responses to use `?`, so that the client has to try and see.  Then returning unimplemented from a handler method would be ok.

Reply:
I clarified this a little but we may want to revisit the design here in a later patch.

Reply:
Without knowing the specific features in question it's hard to say definitively. In general I favor hiding as much of the protocol complexity as possible from users of the crate, but if we can't provide sensible default behavior for certain things then obviously we'll have to have some way for `Handler` implementations to indicate feature support.

Reply:
There are a few, but e.g. we report `QProgramSignals+`.  If the handler then returns "unimplemented" from the corresponding method, gdb will complain, because the server has given inconsistent answers.